syntax = "proto3";

package ftags;

message Command
{
   enum Type
   {
      PING = 0;
      GET_VERSION = 1;
      SHUT_DOWN = 2;
      GET_STATUS = 3;
      FLUSH = 4;
      SAVE_DATABASE = 5;
   }

   Type type = 1;

   string source = 2;
}

message Status
{
   enum Type
   {
      UNKNOWN = 0;
      STARTING = 1;
      IDLE = 2;
      SHUTTING_DOWN = 5;
   }

   Type type = 1;

   string timestamp = 2;
}

message IndexRequest
{
   string projectName = 1;
   string fileName = 2;
   repeated string argument = 3;
   bool shutdownAfter = 4;
}

